pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,119 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zones  []configv1.DNSZone
+		expect []operatorv1.ConditionStatus
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionFalse},
+		},
+		{
+			name: "when no zones available",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones:  []configv1.DNSZone{{}},
```

Comment:
This is actually a slice with 1 zone that has all zero (empty) fields.  Use `[]configv1.DNSZone{}`.

Reply:
done